components/FlexibleHeader/src/MDCFlexibleHeaderViewController.m

```diff
@@ -241,6 +242,17 @@ - (void)scrollViewWillEndDragging:(UIScrollView *)scrollView
 
 - (void)updateTopLayoutGuide {
   [self.topLayoutGuideTopConstraint setConstant:self.flexibleHeaderViewControllerHeightOffset];
+
+  if (@available(iOS 11.0, *)) {
+    CGFloat deviceTopSafeAreaInset = MDCDeviceTopSafeAreaInset();
+    CGFloat top = self.flexibleHeaderViewControllerHeightOffset-deviceTopSafeAreaInset;
+    if (self.parentViewController.additionalSafeAreaInsets.top != top) {
+      UIEdgeInsets insets = self.parentViewController.additionalSafeAreaInsets;
+      insets.top = top;
+      self.parentViewController.additionalSafeAreaInsets = insets;
```

Comment:
This approach sadly doesn't work with the way the FlexibleHeader is built today, as it breaks whenever the trackingScrollView's `contentInsetAdjustmentBehavior` is not set to `never`. Scroll views automatically adjust the content offset to account for the safe area, and the FlexibleHeader responds to changes to the content offset to set its own height. So adding additional safe area insets results in a endless loop that keeps increasing the content offset and therefore the height of the FlexibleHeader. I'll try to find a way to prevent this from happening.

Reply:
Yeah, this is what I was finding. But I didn't have time to really delve into the details or alternative approaches. I think I'll continue to use the old deprecated API topLayoutGuide until this is sorted out.

Reply:
Thoughts @jverkoey? It looks like we would need to change how we calculate the height of the header. Right now it's `-[self fhv_contentOffsetWithoutInjectedTopInset]`, but that doesn't work if we set additional safe area insets.

Reply:
I'd have to dive in to how the autolayout guides are expected to work a bit more before I could have thoughts here. Your gut's probably right @chuga though that we'll need some additional logic in the header's height if we are double counting any offset/height somewhere.